src/command/root-command.ts

```diff
@@ -12,17 +14,31 @@ export class RootCommand {
   @ExternalOption('bee-api-url')
   public beeApiUrl!: string
 
+  @ExternalOption('config-folder')
+  public configFolder!: string
+
   // CLASS FIELDS
 
+  public appName = 'swarm-cli'
+
   public bee!: Bee
 
+  public commandConfig!: CommandConfig
+
   /**
    * Init Root command fields
    *
    * if BEE_API_URL environment variable has been set the CLI will use that connection string
    */
   protected init(): void {
-    this.beeApiUrl = this.beeApiUrl === defaultBeeApiUrl ? process.env.BEE_API_URL || this.beeApiUrl : this.beeApiUrl
+    this.commandConfig = new CommandConfig(this.appName, this.configFolder)
```

Comment:
I find the logic here a bit difficult to follow. If I understand correctly the `CommandConfig` constructor is either
- creates a config file (if there were none) with the default values (but potentially overridden with options)  and returns that 
- or it reads the config file with the values in it

If there is no config it creates a config (for example with the `beeApiUrl`) with the value of the provided option. Therefore in the subsequent invocations it will always use the value of the first invocation unless you pass in an option. But it seems it won't be saved, so the only way to change that is by editing the config file, right?

Reply:
Regardless of the first passed `beeApiUrl` option, it puts the `beeApiUrl`'s default fallback value to the default config file.

After your comment I was thinking about it, and maybe would be better to rearrange the assigment priority in the following way:
1. passed cli option value
2. env variable
3. config
4. option's default value

I would keep generate the default config with the default values of cli options that the cli anyway would fallback to. It is useful to indicate what kind of variables could be set via config, and how.